packages/react-integration/demo-app-ts/src/components/demos/TableDemo/TableCompoundExpandableDemo.tsx

```diff
@@ -196,20 +196,23 @@ export class TableCompoundExpandableDemo extends React.Component<TableProps, Tab
 
   onExpand(event: React.MouseEvent, rowIndex: number, colIndex: number, isOpen: boolean) {
     const newRows = Array.from(this.state.rows);
-    const rowCells = newRows[rowIndex].cells;
+    const rowCells = Array.from(newRows[rowIndex].cells);
+    const thisCell = (rowCells as ICell[])[colIndex];
 
     if (!isOpen) {
       // set all other expanded cells false in this row if we are expanding
-      (rowCells as ICell[]).forEach(cell => {
+      (rowCells as ICell[]).forEach((cell, i) => {
         if (cell.props) {
-          cell.props.isOpen = false;
+          rowCells[i] = { ...cell, props: { ...cell.props, isOpen: false } };
```

Comment:
I thought this too, but I realized through testing that a cloned array still contains references to the original array (it's not a deep clone). Cloning the array only protects against adding/removing/reassigning elements affecting the original array, but if you're modifying properties on an array element, it will still modify the original referenced object in the old array. See my comments here: https://github.com/patternfly/patternfly-react/issues/3762#issuecomment-957958046

I'm not sure if there is a cleaner way to do this, but I was just trying to make sure to be as safe as possible when dealing with cloned data.

Reply:
ah, TIL, thanks.